src/cli/tc.rs

```diff
@@ -1,29 +1,40 @@
-use std::fs;
+use std::collections::{BTreeMap, HashMap};
 use std::path::Path;
 
 use anyhow::Result;
 use clap::Parser;
 use convert_case::{Case, Casing};
 use dotenvy::dotenv;
-use inquire::Confirm;
 use lazy_static::lazy_static;
-use stripmargin::StripMargin;
+use serde::{Deserialize, Serialize};
+use serde_yaml::Value;
 
 use super::command::{Cli, Command};
 use super::generator::Generator;
 use super::update_checker;
 use crate::cli::fmt::Fmt;
+use crate::cli::runtime::{confirm_and_write, create_directory, select_prompt};
 use crate::cli::server::Server;
 use crate::cli::{self, CLIError};
 use crate::core::blueprint::Blueprint;
 use crate::core::config::reader::ConfigReader;
+use crate::core::config::{Config, Expr, Field, RootSchema, Source, Type};
 use crate::core::http::API_URL_PREFIX;
 use crate::core::print_schema;
 use crate::core::rest::{EndpointSet, Unchecked};
+use crate::core::runtime::TargetRuntime;
+
 const FILE_NAME: &str = ".tailcallrc.graphql";
 const YML_FILE_NAME: &str = ".graphqlrc.yml";
 const JSON_FILE_NAME: &str = ".tailcallrc.schema.json";
 
+#[derive(Serialize, Deserialize)]
+pub struct GraphQLRC {
+    schema: Vec<String>,
+    #[serde(flatten)]
+    extra: HashMap<String, Value>,
+}
```

Comment:
Struct should match the shape of the actual value.

```rust
pub struct GraphQLRC {
    schema: Vec<String>
    // Add the remaining fields as specified here:
    // https://the-guild.dev/graphql/config/docs/user/usage
}
```

Reply:
Other option is - Read the current config as `GraphQLRC` (if it exists), and then update it with the new schema files. Before writing it back, take the current config and parse it as `serde_json::Value`. Perform a deep merge between the parsed rc file and the newly generated rc file.

Reply:
created a type with all the fields and doing a deep merge to combine the created `GraphQLRC` object and the object parsed from `.graphqlrc.yml` if it exists, otherwise writing the created object to the file